daemon/k8s_watcher.go

```diff
@@ -670,37 +643,32 @@ func (d *Daemon) addK8sSVCs(svc types.K8sServiceNamespace, svcInfo *types.K8sSer
 		uniqPorts[fePort.Port] = false
 
 		if fePort.ID == 0 {
-			feAddr, err := types.NewL3n4Addr(fePort.Protocol, svcInfo.FEIP, fePort.Port)
-			if err != nil {
-				log.Errorf("Error while creating a new L3n4Addr: %s. Ignoring service...", err)
-				continue
-			}
+			feAddr, _ := types.NewL3n4Addr(fePort.Protocol, svcInfo.FEIP, fePort.Port)
 			feAddrID, err := PutL3n4Addr(*feAddr, 0)
 			if err != nil {
-				log.Errorf("Error while getting a new service ID: %s. Ignoring service %v...", err, feAddr)
+				log.Errorf("Error while getting service ID for %s: %s. Ignoring port %v", feAddr, err, fePort)
 				continue
 			}
-			log.Debugf("Got feAddr ID %d for service %+v", feAddrID.ID, svc)
+			log.Debugf("Resolved service ID %d for service %+v", feAddrID.ID, svc)
 			fePort.ID = feAddrID.ID
 		}
 
 		besValues := []types.LBBackEnd{}
 
 		if k8sBEPort != nil {
-			for epIP := range se.BEIPs {
-				bePort := types.LBBackEnd{
-					L3n4Addr: types.L3n4Addr{IP: net.ParseIP(epIP), L4Addr: *k8sBEPort},
-					Weight:   0,
+			for _, epIP := range se.BEIPs {
+				if epIP.To4() != nil && !isSvcIPv4 {
+					continue
 				}
-				besValues = append(besValues, bePort)
+
+				besValues = append(besValues, types.LBBackEnd{
+					L3n4Addr: types.L3n4Addr{IP: epIP, L4Addr: *k8sBEPort},
+					Weight:   0,
+				})
 			}
 		}
 
-		fe, err := types.NewL3n4AddrID(fePort.Protocol, svcInfo.FEIP, fePort.Port, fePort.ID)
-		if err != nil {
-			log.Errorf("Error while creating a New L3n4AddrID: %s. Ignoring service %v...", err, svcInfo)
-			continue
-		}
+		fe, _ := types.NewL3n4AddrID(fePort.Protocol, svcInfo.FEIP, fePort.Port, fePort.ID)
```

Comment:
This will get redone with the updates k8s state code so I will expect this to become cleaner.